components/nvs/include/nvs.hpp

```diff
@@ -70,6 +70,225 @@ const NvsErrCategory theNvsErrCategory{};
 } // namespace
 
 namespace espp {
+class NVSHandle : public BaseComponent {
```

Comment:
Please move this class into `nvs_handle_espp.hpp` in this folder. This will better enable users to only include the code they want, and will enable a future refactor where the `espp::Nvs` class uses `espp::NvsHandle` internally to reduce code duplication.

Additionally, please add a docstring to the class itself as well, similar to the nvs class above, and include a section/snippet pointing to your new example.

This move to a file will require you to update the `doc/Doxyfile` and `doc/en/nvs.rst` to include the new file.